lib/nostrum/cache/channel_cache.ex

```diff
@@ -6,11 +6,10 @@ defmodule Nostrum.Cache.ChannelCache do
   use GenServer
 
   alias Nostrum.Cache.Guild.GuildServer
-  alias Nostrum.Struct.Guild.Channel
-  alias Nostrum.Struct.DMChannel
+  alias Nostrum.Struct.Channel
   alias Nostrum.Util
 
-  @type channel :: Channel.t() | DMChannel.t()
+  @type channel :: Channel.dm_channel()
```

Comment:
I don't think this is correct. In `get/1` for example, the spec states we return this type when it could be any time of channel. I think this should be removed and the references to a `channel` type in this module should instead use the struct `t/0` type or some other appropriate type.

Reply:
Right now, the `ChannelCache` only stores DM Channels. This part of the Dispatch module is responsible for this:

https://github.com/skippi/nostrum/blob/0e2124b8190fd7d7e7bd5eb05a9d98e1bca14061/lib/nostrum/shard/dispatch.ex#L38

Therefore, I think `Channel.dm_channel()` is fairly accurate for describing the channels that `ChannelCache` works with. Though, I do think that the `ChannelCache` has room for improvement in the future (such as working for all kinds of channels).

Reply:
The `ChannelCache` does currently work for all kinds of channels which is why I brought this up